SourceryFramework/Sources/Parsing/FileParser.swift

```diff
@@ -310,19 +312,27 @@ public final class FileParser {
 // MARK: - Details parsing
 extension FileParser {
 
-    fileprivate func parseTypeRequirements(_ dict: [String: SourceKitRepresentable]) -> (name: String, kind: SwiftDeclarationKind, accessibility: AccessLevel)? {
-        guard let kind = (dict[SwiftDocKey.kind.rawValue] as? String).flatMap({ SwiftDeclarationKind(rawValue: $0) }),
-              var name = dict[SwiftDocKey.name.rawValue] as? String else { return nil }
-
-        if case .enumelement = kind, let colon = name.firstIndex(of: "(") {
-            name = String(name[..<colon])
-        }
-
-        if extract(.name, from: dict)?.hasPrefix("`") == true {
-            name = "`\(name)`"
+    fileprivate func parseTypeRequirements(_ dict: [String: SourceKitRepresentable]) -> (name: String?, kind: SwiftDeclarationKind, accessibility: AccessLevel)? {
+        guard let kind = (dict[SwiftDocKey.kind.rawValue] as? String).flatMap({ SwiftDeclarationKind(rawValue: $0) }) else {
+            return nil
         }
 
         let accessibility = (dict["key.accessibility"] as? String).flatMap({ AccessLevel(rawValue: $0.replacingOccurrences(of: "source.lang.swift.accessibility.", with: "") ) }) ?? .none
+        
+        let name: String? = (dict[SwiftDocKey.name.rawValue] as? String).map {
+            var name: String = $0
+            
+            if case .enumelement = kind, let colon = name.firstIndex(of: "(") {
```

Comment:
why does it search for first `(` while called `colon`?

Reply:
I don't know. The person who touched this code was @Liquidsoul (in 2019).

I renamed it to `openingBrace`? Does it suit?